client-react/src/pages/app/functions/function/function-log/FunctionLogAppInsightsDataLoader.tsx

```diff
@@ -258,13 +267,12 @@ const FunctionLogAppInsightsDataLoader: React.FC<FunctionLogAppInsightsDataLoade
     setAllLogEntries([]);
   };
 
-  const getShowFilteredLogsMessage = useCallback(() => {
-    return (
-      Url.isFeatureFlagEnabled(CommonConstants.FeatureFlags.useNewFunctionLogsApi) &&
-      functionsRuntimeVersion === CommonConstants.FunctionsRuntimeVersions.four &&
-      !!liveLogsSessionId
-    );
-  }, [functionsRuntimeVersion, liveLogsSessionId]);
+  const useNewFunctionLogsApi = useCallback(() => {
+    const isNewLogsApiEnabled =
+      Url.isFeatureFlagEnabled(CommonConstants.FeatureFlags.useNewFunctionLogsApi) || !!isFunctionLogsApiFlightingEnabled;
+
+    return isNewLogsApiEnabled && functionsRuntimeVersion === CommonConstants.FunctionsRuntimeVersions.four;
+  }, [functionsRuntimeVersion, liveLogsSessionId, isFunctionLogsApiFlightingEnabled]);
```

Comment:
Consider memoizing this to store this as a value rather than a function which returns the value. Add `useMemo` to the `react` import in line 1.

```suggestion
  const useNewFunctionLogsApi = useMemo(() => {
    const isNewLogsApiEnabled =
      Url.isFeatureFlagEnabled(CommonConstants.FeatureFlags.useNewFunctionLogsApi) || !!isFunctionLogsApiFlightingEnabled;

    return isNewLogsApiEnabled && functionsRuntimeVersion === CommonConstants.FunctionsRuntimeVersions.four;
  }, [functionsRuntimeVersion, liveLogsSessionId, isFunctionLogsApiFlightingEnabled]);
```